File: WebApplication/java/src/TestServer.java
import java.io.*;
import java.net.*;

public class TestServer {
	public static void main (String[] args) throws Exception {
		try (ServerSocket server = new ServerSocket(8001);
				FileOutputStream fos = new FileOutputStream("server_recv.txt")) {
			System.out.println("Waiting a connection from a client...");
			Socket socket = server.accept();
			System.out.println("Connect from a client.");

			int ch;
			InputStream input = socket.getInputStream();
			while ((ch = input.read()) != -1) {
				fos.write(ch);
			}
			socket.close();
		} catch (Exception e) {
			e.printStackTrace();
		}
	}
}

File: WebApplication/go/test_server.go
package main

import (
	"bufio"
	"fmt"
	"net"
	"os"
)

func waitClient(listener net.Listener) {
	fmt.Println("Waiting a connection from a server...")
	conn, err := listener.Accept()
	if err != nil {
		panic(err)
	}
	fmt.Println("Connect from a client.")

	go goEcho(conn)
	waitClient(listener)
}

func goEcho(conn net.Conn) {
	defer conn.Close()
	echo(conn)
}

func echo(conn net.Conn) {
	// var buf = make([]byte, 1024)

	/*
		_, err := conn.Read(buf)
		if err != nil {
			if err == io.EOF {
				return
			} else {
				panic(err)
			}
		}
	*/

	fp, err := os.Create("server_recv.txt")
	if err != nil {
		fmt.Println("File error.")
		return
	}
	defer fp.Close()

	output := bufio.NewWriter(fp)
	n, err := output.ReadFrom(conn)
	if err != nil {
		fmt.Println("Output error.")
		return
	}
	fmt.Println(n)
}

func main() {
	listener, err := net.Listen("tcp", "localhost:8001")

	if err != nil {
		panic(err)
	}

	waitClient(listener)
}
